extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -336,11 +352,27 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     // TODO: Check for required and allowed parameters. Probably use
     // validateData, from 3p/3p/js, after noving it someplace common.
     const startTime = Date.now();
-    return getPageLevelParameters_(this.win, this.getAmpDoc(), startTime)
-        .then(pageLevelParameters =>
-        googleAdUrl(this, DOUBLECLICK_BASE_URL, startTime,
-            Object.assign(this.getBlockParameters_(), pageLevelParameters),
-          ['108809080']));
+
+    const pageLevelParametersPromise = getPageLevelParameters_(
+        this.win, this.getAmpDoc(), startTime);
+    const rtcRequestPromise = this.executeRtc_();
+    return Promise.all(
+      [pageLevelParametersPromise, rtcRequestPromise]).then(values => {
+        const pageLevelParameters = values[0];
+        const parameters = Object.assign(
+            this.getBlockParameters_(), pageLevelParameters);
+        if (values[1]) {
+          const rtcTotalTime = values[1]['rtcTotalTime'];
+          const rtcSuccess = values[1]['success'];
+          if (rtcTotalTime) {
+            parameters['artc'] = rtcTotalTime;
+            parameters['ard'] = parseUrl(rtcConfig['endpoint']).hostname;
```

Comment:
I'm thinking a cleaner interface is to just have executeRtc return parameters to be added.